puppet-mode.el

```diff
@@ -632,6 +633,40 @@ of the initial include plus puppet-include-indent."
         (indent-line-to 0)))))
 
 
+;;;; SMIE
+
+(defvar puppet-use-smie nil
+  "Whether to use SMIE or not.")
```

Comment:
The problem is that it doesn't really work so it should be opt-in and flipping a variable is easier than switching branches. Eventually, if it ever works, sure you don't need a variable, but until then...